Zenodo.js

```diff
@@ -0,0 +1,309 @@
+	GNU Affero General Public License for more details.
+	
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+    if (url.indexOf('/record/')>-1) {
+    	var collections = ZU.xpath(doc, '//div[contains(@class, "metadata")]//a[@class="navtrail"]');
+    	for (var i=0; i<collections.length; i++) {
+    		//Z.debug(collections[i].textContent);
+    		switch (collections[i].textContent) {
+    			case "Presentations":
+    				return "presentation";
+    			case "Software":
+    				return "computerProgram";
+    			case "Videos/Audio":
+    				return "videoRecording";//or audioRecording?
+    			case "Images":
+    				return "artwork";
+    			case "Presentations":
+    				return "presentation";
+    			case "Posters":
+    				return "";
+    			case "Lessons":
+    			case "Books":
+    				return "book";
+    			case "book-section":
+    				return "chapter";
+    			case "conference-papers":
+    				return ""
+    			case "patents":
+    				return "patent";
+    			case "reports":
+    			case "working-papers":
+    			case "technical-notes":
+    			case "project-deliverables":
+    			case "project-milestones":
+    			case "proposals":
+    				return "report";
+    			case "Theses":
+    				return "thesis";
+    			case "Datasets":
+    				//change when dataset as itemtype is available
+    			case "journal-articles":
+    			case "preprints":
+    				return "journalArticle";
+    		}
+    	}
+    	return "journalArticle";
+    } else if (getSearchResults(doc, true)) {
+        return "multiple";
+    }
+}
+
+function getSearchResults(doc, checkOnly) {
+    var items = {};
+    var found = false;
+    var rows = ZU.xpath(doc, '//div[contains(@class, "row")]//h4/a');
+    for (var i=0; i<rows.length; i++) {
+        var href = rows[i].href;
+        var title = ZU.trimInternal(rows[i].textContent);
+        if (!href || !title) continue;
+        if (checkOnly) return true;
+        found = true;
+        items[href] = title;
+    }
+    return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+    if (detectWeb(doc, url) == "multiple") {
+        Zotero.selectItems(getSearchResults(doc, false), function (items) {
+            if (!items) {
+                return true;
+            }
+            var articles = new Array();
+            for (var i in items) {
+                articles.push(i);
+            }
+            ZU.processDocuments(articles, scrape);
+        });
+    } else {
+        scrape(doc, url);
+    }
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	var type = detectWeb(doc, url);
+    // Embedded Metadata
+    translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+    translator.setDocument(doc);
+
+    /*translator.setHandler('itemDone', function (obj, item) {
+        //
```

Comment:
I'd like to keep the DOIs where they're not in the Zotero data model, how about something like:
```
if (item.DOI && type != "journalArticle" && type != "conferencePaper") {
  item.extra = "DOI: " + item.DOI;
}
```

also, do you want to code the dataset item type into extra?